backend/infrahub/core/migrations/graph/m019_restore_rels_to_time.py

```diff
@@ -0,0 +1,78 @@
+
+        return result
+
+    async def execute(self, db: InfrahubDatabase) -> MigrationResult:
+        """
+        Fix corrupted state introduced by Migration012 when duplicating a CoreAccount (branch Aware)
+        being part of a CoreStandardGroup (branch Agnostic). Database is corrupted at multiple points:
+        - Old CoreAccount node <> group_member node `active` edge has no `to` time (possibly because of #5590).
+        - Old CoreAccount node <> group_member node `deleted` edge is on `-global-` branch instead of `main`.
+        - New CoreAccount node <> group_member node `active` edge is on `-global-` branch instead of `main`.
+
+        Also, users having deleted corresponding CoreStandardGroup will also have the following data corruption,
+        as deletion did not happen correctly due to above issues:
+        - Both new CoreAccount node <> group_member node and CoreStandardGroup node <> group_member node edges
+          have not been deleted (ie status is `active` without `to` time and no additional `deleted` edge).
+
+        This migration fixes all above issues to have consistent edges, and fixes IFC-1204.
+        """
+
+        query = """
+            // HAS_ATTRIBUTE links are fine because the notion of "aware" or "agnostic" is only for relationships
+            MATCH (a: CoreAccount)-[r:IS_RELATED|IS_PROTECTED|IS_VISIBLE]-(rel: Relationship)-[r3]-(c:CoreGenericAccount)
+            WITH a, rel, COUNT(rel) AS rel_count, c, r3
+            WHERE rel_count = 2 and rel.branch_support = 'aware'
+            MATCH (a)-[r1]->(rel)
+            MATCH (a)-[r2]->(rel)
+            WHERE r1.status = 'deleted' AND r1.branch = '-global-'
+            AND r2.status <> 'deleted' AND r2.branch <> '-global-' AND r2.to IS NULL
+            AND c <> a AND r3.branch = '-global-'
+
+            // Fixing missing `to` on non-global edge
+            SET r2.to = r1.from
+
+            // Fix branch of the new edge
+            SET r3.branch = r2.branch
+
+            // Remove `deleted` edge as it is not on main branch.
+            DELETE r1
+
+            WITH a, c, rel, r3
+            OPTIONAL MATCH (rel)-[r4:IS_RELATED]-(core_standard_group: CoreStandardGroup)-[is_part_of: IS_PART_OF]->(root: Root)
+
+            WHERE is_part_of.to IS NOT NULL
+            AND r3.to IS NULL AND r3.status = 'active' AND is_part_of.branch = r3.branch
+            AND r4.to IS NULL AND r4.status = 'active' AND is_part_of.branch = r4.branch
+            SET r3.to = is_part_of.to
+            SET r4.to = is_part_of.to
+            RETURN a, c, rel, core_standard_group, root
+            // maybe make sure to time is higher than from time of r3/r4 but if it was not the case if means CoreStdGroup would have been deleted
+            // beore kind migrating and thus we would not match all of these anyway
+
+        """
+
+        await db.execute_query(query=query, name="update_relationships_to")
+
+        return MigrationResult()
```

Comment:
I will probably update this migration to make the new `CoreAccount` node introduced by `Migration012` to inherit from `CoreNode` as discussed.